shared/src/main/scala/coop/rchain/store/KeyValueStoreManagerSyntax.scala

```diff
@@ -15,10 +15,24 @@ final class KeyValueStoreManagerOps[F[_]: Sync](
     // KeyValueStoreManager extensions / syntax
     private val manager: KeyValueStoreManager[F]
 ) {
+
+  /**
+    * Returns typed key-value store (DB).
+    *
+    * @param name database name
+    * @param kCodec codec for the key
+    * @param vCodec codec for the value
+    */
   def database[K, V](
       name: String,
       kCodec: Codec[K],
       vCodec: Codec[V]
   ): F[KeyValueTypedStore[F, K, V]] =
     manager.store(name).map(_.toTypedStore(kCodec, vCodec))
+
+  /**
+    * Wraps manager with Resource calling shutdown on exit.
+    */
+  def asResource: Resource[F, KeyValueStoreManager[F]] =
+    Resource.make(manager.pure[F])(_.shutdown)
```

Comment:
Potentially, we can remove _shutdown_ function from the manager interface and leave to implementation to be wrapped in Resource.